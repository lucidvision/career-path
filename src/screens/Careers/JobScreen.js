import React, { Component } from 'react'
import { Text, View } from 'react-native'

class JobScreen extends Component {
  render () {
    const { params } = this.props.navigation.state
    return (
      <View>
        <Text>{params.name}</Text>
        <Text>{`Salary: ${params.salary}`}</Text>
        <Text>{`Education: ${params.education}`}</Text>
      </View>
    )
  }
}

export default JobScreen
