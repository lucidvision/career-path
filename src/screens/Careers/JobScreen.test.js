import React from 'react'
import { shallow } from 'enzyme'
import JobScreen from './JobScreen'

const createJobScreen = (override) => {
  const props = {
    navigation: {
      state: {
        params: {
          name: 'Doctor',
          salary: 100000,
          education: 'MD'
        }
      }
    },
    ...override
  }

  return shallow(<JobScreen {...props} />)
}

describe('JobScreen', () => {
  const component = createJobScreen()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })
})
