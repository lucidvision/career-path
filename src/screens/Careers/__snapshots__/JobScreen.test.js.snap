// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobScreen renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Doctor
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Salary: 100000
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Education: undefined
  </Text>
</View>
`;
